markdown_it/rules_inline/text.py

```diff
@@ -34,13 +36,15 @@
     "}",
     "~",
 }
+_RE_TERMINATOR_CHAR = re.compile("[" + re.escape("".join(_TerminatorChars)) + "]")
```

Comment:
I prefer to do regex compilations lazily, so just installing / CLI start-up is fast (note python caches `re.compile` internally)

Reply:
Unfortunately [`re.search`](https://docs.python.org/3/library/re.html#re.search) does not give access to start `pos`  unlike [`Pattern.search`](https://docs.python.org/3/library/re.html#re.Pattern.search)

Reply:
Ah maybe I misunderstood. If what you mean is to keep the `re.compile` call but move it inside `def text` scope then yes that'll work. But this is such hot code that it has a significant performance impact (raises `Min` value in `bench-core` to ~76.7 from ~74.7 on my machine).

Reply:
or just use a get function with `@cache`

Reply:
perfect thanks